Sming/Arch/Esp8266/build.mk

```diff
@@ -4,7 +4,8 @@
 #
 ##############
 
-CFLAGS				+= -DARCH_ESP8266
+CPPFLAGS			+= -DARCH_ESP8266
+CFLAGS 				+= -std=c11
```

Comment:
> Should C coding standard be set in main build.mk file, rather than repeating for arch. ?

We could set a default one in the main build.mk file but allow every architecture to change it if needed. For Arch Esp32 we need c99 standard for the c files otherwise too may problems pop-up.